examples/src/java/org/pantsbuild/example/page.md

```diff
@@ -107,6 +107,13 @@ end-at=*substring*<br>
 When excerpting the file to include, stop at a line containing
 *substring*.
 
+lines=*start-line*-*end-line*<br>
```

Comment:
It would be worth mentioning here that you should always prefer to use the `start/end` style parameters, and that the `lines` parameter is only for use where comments aren't legal.

But... as far as I can, comments are in fact allowing in `requirements.txt` files? Maybe not in practice...?

Reply:
That's right, and I didn't mean to imply comments aren't legal (on their own line) or are an awful practice in `requirements.txt`. In the vein of "making it harder to break documentation by accident", using `lines` seems more robust and understandable than `start`/`end` for `requirements.txt`, because:

1. using `start`/`end` to include `requirements.txt` relies on the file using a package with a specific name, unless you use comments to mark the section to include.  But if `requirements.txt` is generated, e.g. with `pip freeze`, you have to remember to add the comments back when using `start`/`end`.
2.  *usually* there's no specific section of the file we'd want to draw attention to -- e.g. when we `!inc` this repo's `requirements.txt`, we only care that it shows some section of the file and that it's not too long. If we used comment markers and `start`/`end`, the amount of the file included in the docs might silently become much longer just by adding packages. This isn't "breaking the docs", but it's probably not desired.

One situation where `lines` should *not* be used is if we *wanted* to include a specific section, for example if we knew that a specific update of some package we use has a bug. In that case, we could add a comment above the line of `requirements.txt` which pins that package so that nobody bumps the version. If we wanted to include that specific part of `requirements.txt` in documentation, we could and definitely should use `start`/`end` so that we could make other changes to the file without breaking the docs.

I'll describe when `lines` vs `start`/`end` should be used and why in more depth here.

Reply:
> using lines seems more robust and understandable than start/end for requirements.txt, because:

Maybe for that case. But generally you _are_  trying to render a particular codeblock into the documentation, and that case will break constantly in the context of lines.

> usually there's no specific section of the file we'd want to draw attention to -- e.g. when we !inc this repo's requirements.txt,

Again, maybe for this case. But if I'm trying to render a method body (for example), or a target definition from a BUILD file, edits before those lines in the file would silently break my `inc`.

There are very specific cases where lines are ok, but in cases where comments are legal, they should be used.